packages/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt

```diff
@@ -0,0 +1,90 @@
+package expo.modules.cellular
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.sip.SipManager
+import android.os.Build
+import android.telephony.TelephonyManager
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.ModuleRegistryDelegate
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.RegistryLifecycleListener
+import java.util.*
+
+
+class CellularModule(
+  private val mContext: Context,
+  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate(),
+) : ExportedModule(mContext), RegistryLifecycleListener {
+  override fun getName(): String = "ExpoCellular"
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    moduleRegistryDelegate.onCreate(moduleRegistry)
+  }
+
+  override fun getConstants(): HashMap<String, Any?> {
+    val constants = HashMap<String, Any?>()
+    constants["allowsVoip"] = SipManager.isVoipSupported(mContext)
+    val telephonyManager = mContext.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+    constants["isoCountryCode"] = telephonyManager.simCountryIso
+
+    //check if sim state is ready
+    if (telephonyManager.simState == TelephonyManager.SIM_STATE_READY) {
+      constants["carrier"] = telephonyManager.simOperatorName
+      constants["mobileCountryCode"] = telephonyManager.simOperator.substring(0, 3)
+      constants["mobileNetworkCode"] = StringBuilder(telephonyManager.simOperator).delete(0, 3).toString()
+    } else {
+      constants["carrier"] = null
+      constants["mobileCountryCode"] = null
+      constants["mobileNetworkCode"] = null
+    }
+    return constants
+  }
+
+  @SuppressLint("MissingPermission")
+  @ExpoMethod
+  fun getCellularGenerationAsync(promise: Promise) {
+    try {
+      val mSystemService = mContext.getSystemService(Context.TELEPHONY_SERVICE)
+      val mTelephonyManager = if (mSystemService != null) {
+        mSystemService as TelephonyManager
```

Comment:
I'm not sure why the Java code had the `m` prefix for these local variables (perhaps in an earlier version of the module's first PR they were member variables and were not cleaned up properly) but we can safely remove it here.
```suggestion
      val systemService = mContext.getSystemService(Context.TELEPHONY_SERVICE)
      val telephonyManager = if (systemService != null) {
        systemService as TelephonyManager
```